hugr/src/ops.rs

```diff
@@ -301,12 +300,22 @@ macro_rules! impl_op_name {
 
 use impl_op_name;
 
+/// Marker for the [`OpName`] wrapper.
+#[doc(hidden)]
+pub enum OpNameMarker {}
+
+/// A unique identifier for a operation.
+pub type OpName = string_newtype::SmolStrBuf<OpNameMarker>;
+
+/// Slice of a [`OpName`] operation identifier.
+pub type OpNameSlice = string_newtype::SmolStrRef<OpNameMarker>;
+
```

Comment:
@doug-q Requesting a second opinion here. This looks a bit boilerplate-y/confusing, but it may reduce the amount of code on the callsites.

An alternative is to drop `OpNameSlice` and just use references to cloned `OpName`s for parameters.